### PR TITLE
ignite(phase-0): detect partial RFC state and resume from first missing sub-phase

### PR DESCRIPTION
**Pull request by JMBattista**
Teach smithy.ignite Phase 0 to classify any existing `.rfc.md` it encounters
as `fresh`, `partial`, or `complete` before running the audit/review loop,
and route each state to a distinct next step. Partial files resume at the
first missing sub-phase (computed from a section-to-sub-phase map covering
3a–3f) after user confirmation, without re-running completed sub-phases.
Fresh header-only files skip the review loop and hand off to 3a with the
existing file preserved in place. Complete files continue into the existing
Phase 0a–0b audit as before.

Phase 3's append-and-continue preamble now carries an explicit resume note
that skips earlier sub-phases, leaves the accumulating file untouched on
entry, and relies on the existing `rfc_file_path` parameter to bridge prior
sections into dispatched sub-agents. The Routing section delegates the
review-vs-resume-vs-new decision to Phase 0, so the mid-intake redirect no
longer asks the user inline. Edge cases for partial-from-a-different-idea
and harmonize-crash mid-rewrite are handled explicitly.

Template tests assert the renamed Phase 0 heading, the three state labels,
the first-missing-sub-phase wiring, and the Phase 3 resume note all render
in the claude-variant ignite prompt, and that none of this bleeds into the
default (non-agent) variant.

https://claude.ai/code/session_0143LJZ6nK3Pnj6Tu7tNbcTh

### CODE REVIEW

File: src/templates/agent-skills/commands/smithy.ignite.prompt [hunk 1]
```diff
@@ -27,30 +27,116 @@ If no input is clear from the above, ask the user what idea they want to worksho
...
+{{#ifAgent}}
+## Phase 0: State Detection and Review Loop
+{{else}}
 ## Phase 0: Review Loop
+{{/ifAgent}}
 
 Triggered when:
 - The input explicitly points to an existing `.rfc.md` file, **or**
 - Phase 1 detected a matching RFC folder during the `docs/rfcs/` scan and the
   user chose to review it (see Routing above).
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply src/templates/agent-skills/commands/smithy.ignite.prompt]: Fixed in 7102dc7: updated the "Triggered when" bullet to match the new Routing — Phase 1 now hands control to Phase 0 with the matched `.rfc.md` (no inline review-vs-create-new ask).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/templates/agent-skills/commands/smithy.ignite.prompt [hunk 2]
```diff
@@ -182,6 +268,32 @@ append-and-continue protocol for sub-phases 3a–3f. For inline sub-phase 3e, th
...
+3. **Begin dispatch** from the designated starting sub-phase and continue
+   through sub-phase 3g normally, using the append-and-continue protocol
+   above for each remaining sub-phase.
+
+No extra context bridging is required for resumed sub-phases. Each sub-phase
+below already passes `rfc_file_path` — the path to the accumulating
+`<slug>.rfc.md` — to its dispatched sub-agent, so every section written by an
+earlier sub-phase (whether in this session or in an interrupted prior
+session) flows into the current sub-agent via the existing disk-read
+contract.
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply src/templates/agent-skills/commands/smithy.ignite.prompt]: Fixed in 7102dc7: the Resume Hand-off note now calls out the sub-phase 3a `rfc_file_path` exception explicitly. When Phase 0 hands off into 3a (either a `fresh` header-only file or a `partial` file where 3a is somehow first missing), the orchestrator MUST pass `rfc_file_path` to smithy-prose so it can see the on-disk state before writing Summary + Motivation, even though the normal fresh pipeline omits it at 3a.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/templates/agent-skills/commands/smithy.ignite.prompt [hunk 3]
```diff
@@ -27,30 +27,116 @@ If no input is clear from the above, ask the user what idea they want to worksho
...
-- If the user chooses to review, go to **Phase 0: Review Loop** with that `.rfc.md`.
-- If the user chooses to create new, continue Phase 1 with the next available `NNN`.
+folders. If a folder's slug is a close match to the derived slug for the new
+idea (e.g., `docs/rfcs/2026-001-plugin-system/` already exists when the user
+asks to "build a plugin system"), **stop intake** and hand control to
+**Phase 0** with the matched `.rfc.md`. Do **not** ask the user "review or
+create new" inline here — that decision is delegated to Phase 0's
+state-detection step, which is now the single place that handles review,
+resume, and new-RFC branching. The option to **create a new RFC instead** of
+touching the existing file remains available to the user from inside Phase 0's
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply src/templates/agent-skills/commands/smithy.ignite.prompt:50]: Fixed in 7102dc7: bifurcated the Routing section with `{{#ifAgent}}/{{else}}`. The new state-detection delegation language now only renders in the agent variant. The default (non-agent) variant keeps its original "review or create new" inline ask, which matches its non-agent Phase 0 (which has no state-detection branching to delegate to). The default-variant test guards continue to pass.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/templates/agent-skills/commands/smithy.ignite.prompt [hunk 4]
```diff
@@ -182,6 +268,32 @@ append-and-continue protocol for sub-phases 3a–3f. For inline sub-phase 3e, th
...
+   disk (either a header-only `fresh` file or a `partial` file with some
+   sections already written) and is preserved as-is.
+3. **Begin dispatch** from the designated starting sub-phase and continue
+   through sub-phase 3g normally, using the append-and-continue protocol
+   above for each remaining sub-phase.
+
+No extra context bridging is required for resumed sub-phases. Each sub-phase
+below already passes `rfc_file_path` — the path to the accumulating
+`<slug>.rfc.md` — to its dispatched sub-agent, so every section written by an
+earlier sub-phase (whether in this session or in an interrupted prior
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply src/templates/agent-skills/commands/smithy.ignite.prompt]: Fixed in 7102dc7: extended the Phase 3 Resume Hand-off note with an explicit context-reconstruction step. Before dispatching the resumed sub-phase, the orchestrator now MUST rebuild `idea_description` (re-derive from the existing Summary / Motivation sections; fall back to user invocation arguments or ask the user if those sections aren't yet on disk) and `clarify_output` (prefer loading `.clarify-log.md`; if no log is present, run Phase 2 before dispatching rather than silently dispatching with empty clarification). This addresses the gap where resumed sub-phases would otherwise see missing or fabricated intake/clarify context.

---
_Generated by [Claude Code](https://claude.ai/code)_
